File: Src/CSharp/FTP-related/ClientFtp/RemoteListViewItem.cs
using System;
using System.Windows.Forms;
using Xceed.Ftp;

namespace ClientFtp
{
	public class RemoteListViewItem : ListViewItem
	{
		public RemoteListViewItem( FtpItemInfo info )
      : base( info.Name )
		{
      if( info == null )
        throw new ArgumentNullException( "info" );

      m_info = info;

      // We only show the size for a File or unknown.
      if( ( info.Type == FtpItemType.File ) || ( info.Type == FtpItemType.Unknown ) )
      {
        this.SubItems.Add( RemoteFolderTreeNode.FormatSize( info.Size ) );
      }
      else
      {
        this.SubItems.Add( "" );
      }

      this.SubItems.Add( info.DateTime.ToString() );

      switch( info.Type )
      {
        case FtpItemType.File:
        case FtpItemType.Unknown:
          this.ImageIndex = ( int ) FtpItemIconEnum.File;
          break;

        case FtpItemType.Folder:
          this.ImageIndex = ( int ) FtpItemIconEnum.ClosedFolder;
          break;

        case FtpItemType.Link:
          this.ImageIndex = ( int ) FtpItemIconEnum.ClosedLink;
          break;
      }
    }

    public FtpItemInfo Info
    {
      get { return m_info; }
    }

    private FtpItemInfo m_info = FtpItemInfo.Empty;
	}
}

File: Src/Visual Basic .NET/FTP-related/ClientFtp/RemoteListViewItem.vb
Imports System.Windows.Forms
Imports Xceed.Ftp

Public Class RemoteListViewItem
  Inherits ListViewItem

  Public Sub New(ByVal info As FtpItemInfo)
    MyBase.New(info.Name)

    If info Is Nothing Then
      Throw New ArgumentNullException("info")
    End If

    m_info = info

    ' We only show the size for a File or unknown.
    If info.Type = FtpItemType.File Or info.Type = FtpItemType.Unknown Then
      Me.SubItems.Add(RemoteFolderTreeNode.FormatSize(info.Size))
    Else
      Me.SubItems.Add("")
    End If

    Me.SubItems.Add(info.DateTime.ToString())

    Select Case info.Type
      Case FtpItemType.File, FtpItemType.Unknown
        Me.ImageIndex = CInt(FtpItemIconEnum.File)

      Case FtpItemType.Folder
        Me.ImageIndex = CInt(FtpItemIconEnum.ClosedFolder)

      Case FtpItemType.Link
        Me.ImageIndex = CInt(FtpItemIconEnum.ClosedLink)
    End Select
  End Sub


  Public ReadOnly Property Info() As FtpItemInfo
    Get
      Return m_info
    End Get
  End Property


  Private m_info As FtpItemInfo = FtpItemInfo.Empty
End Class
